core/src/main/java/com/sun/mail/util/PropUtil.java

```diff
@@ -149,6 +162,16 @@ private static int getInt(Object value, int def) {
 	return def;
     }
 
+	private static ScheduledExecutorService getScheduledExecutorService(Object value) {
+		if (value == null)
+			return null;
+		if (value instanceof ScheduledExecutorService) {
+			return (ScheduledExecutorService) value;
+		}
+
+		return null;
```

Comment:
@jmehrens btw, wouldn't it be better to throw here an illegal argument exception, what do you think?

Reply:
ClassCastException would also be fine too.  Failing is probably appropriate here. I forgot that we will have to add a test case for wrong type anyway.

Reply:
fixed